### PR TITLE
backend/hls: Initial commit of high-level-synthesis backend

### PR DESCRIPTION
**Pull request by sjalander**

### CODE REVIEW

File: libjlm/Makefile.sub [hunk 1]
```diff
@@ -63,6 +64,26 @@ LIBJLM_SRC = \
...
+	libjlm/src/backend/hls/rvsdg2rhls/add-prints.cpp \
+	libjlm/src/backend/hls/rvsdg2rhls/add-buffers.cpp \
+	\
+	libjlm/src/backend/hls/rhls2firrtl/base-hls.cpp \
+	libjlm/src/backend/hls/rhls2firrtl/dot-hls.cpp \
+	libjlm/src/backend/hls/rhls2firrtl/verilator-harness-hls.cpp \
+	libjlm/src/backend/hls/rhls2firrtl/firrtl-hls.cpp \
+	libjlm/src/backend/hls/rhls2firrtl/mlirgen.cpp \
+	\
+	$(LLVM_VER_SRC) \
```
Comment (phate) [inline libjlm/Makefile.sub:86]: This seems to be nowhere defined!?

File: libjlm/include/jlm/opt/inlining.hpp [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #define JLM_OPT_INLINE_HPP
 
 #include <jlm/opt/optimization.hpp>
+#include <jive/rvsdg.hpp>
+#include <jlm/ir/operators/lambda.hpp>
```
Comment (phate) [inline libjlm/include/jlm/opt/inlining.hpp:11]: I believe that both headers are not necessary here. Forward declaring jive::input, jive::simple_node and lambda::node should be enough

File: libjlm/include/jlm/ir/hls/hls.hpp [hunk 3]
```diff
@@ -0,0 +1,560 @@
+//
+// Created by david on 7/2/21.
+//
```
Comment (phate) [inline libjlm/include/jlm/ir/hls/hls.hpp:3]: Could we bring the headers for all newly added files into the same format as all the other headers, please?

File: libjlm/Makefile.sub (hunk 1 above)
Comment (sjalander) [reply libjlm/Makefile.sub:86]: FIXED: This should have been removed.

File: libjlm/include/jlm/opt/inlining.hpp (hunk 2 above)
Comment (sjalander) [reply libjlm/include/jlm/opt/inlining.hpp:11]: Seems like you assume you are talking to somebody that have a complete overview of the code.
If you meant that you think that jive/rvsdg.hpp is not needed, it has been removed.

File: libjlm/include/jlm/ir/hls/hls.hpp (hunk 3 above)
Comment (sjalander) [reply libjlm/include/jlm/ir/hls/hls.hpp]: The headers have been replaced

**Pull request merged**